src/network/networkd-nexthop.c

```diff
@@ -347,14 +379,16 @@ static void log_nexthop_debug(const NextHop *nexthop, uint32_t id, const char *s
         if (!DEBUG_LOGGING)
                 return;
 
+        if (nexthop->id != id)
+                (void) asprintf(&new_id, "→%"PRIu32, id);
+
         (void) in_addr_to_string(nexthop->family, &nexthop->gw, &gw);
 
-        if (nexthop->id == id)
-                log_link_debug(link, "%s nexthop: id: %"PRIu32", gw: %s, blackhole: %s",
-                               str, nexthop->id, strna(gw), yes_no(nexthop->blackhole));
-        else
-                log_link_debug(link, "%s nexthop: id: %"PRIu32"→%"PRIu32", gw: %s, blackhole: %s",
-                               str, nexthop->id, id, strna(gw), yes_no(nexthop->blackhole));
+        HASHMAP_FOREACH(nhg, nexthop->group)
+                (void) strextendf(&group, "%s%"PRIu32":%"PRIu32, group ? "," : "", nhg->id, nhg->weight+1);
```

Comment:
rework to use new strextendf_with_separator?

Reply:
Yeah!